packages/core/src/main.ts

```diff
@@ -0,0 +1,144 @@
+import {
+  ConnectedStarknetWindowObject,
+  StarknetWindowObject,
+} from "./StarknetWindowObject"
+import discovery, { WalletProvider } from "./discovery"
+import { IStorageWrapper, LocalStorageWrapper } from "./localStorageStore"
+import { pipe } from "./utils"
+import { FilterList, filterBy, filterByPreAuthorized } from "./wallet/filter"
+import { isWalletObj } from "./wallet/isWalletObject"
+import { scanObjectForWallets } from "./wallet/scan"
+import { Sort, sortBy } from "./wallet/sort"
+
+export type {
+  AccountChangeEventHandler,
+  AddStarknetChainParameters,
+  ConnectedStarknetWindowObject,
+  NetworkChangeEventHandler,
+  RpcMessage,
+  StarknetWindowObject,
+  SwitchStarknetChainParameter,
+  WalletEventHandlers,
+  WalletEvents,
+  WatchAssetParameters,
+} from "./StarknetWindowObject"
+export type { WalletProvider } from "./discovery"
+
+export interface SnWalletOptions {
```

Comment:
these options aren't related to the wallet at all, let's call this interface `GetStarknetOptions`/ `GSOptions`?